CorsixTH/Src/persist_lua.cpp

```diff
@@ -270,217 +267,194 @@ class lua_persist_basic_writer : public lua_persist_writer {
       // things).
       lua_getfenv(L, 1);
       lua_pushvalue(L, iIndex);
-      lua_gettable(L, -2);  // Might (indirectly) call writeObjectRaw
-      uint64_t iValue = (uint64_t)lua_tonumber(L, -1);
-      lua_pop(L, 2);
+
+      lua_rawget(L, -2);
+      uint64_t iValue = static_cast<uint64_t>(lua_tonumber(L, -1));
```

Comment:
Old code, but shouldn't this be some `UINT64_MAX` or so?

Reply:
I don't follow, was this comment on the wrong line?

Reply:
`-1` looks like an attempt to write signed data into an unsigned value, in particular, the bit representation of `0xFFFF...`.

Reply:
No, that -1 means the top of the lua stack. Return the item at the top of the lua stack as if it was a number.